File: examples/01_type_error.py
def total_price(items, tax_rate):
    subtotal = sum(item["price"] for item in items)
    return subtotal + tax_rate


def checkout(cart):
    return total_price(cart, "0.08")


if __name__ == "__main__":
    cart = [{"price": 10}, {"price": 20}, {"price": 5}]
    print(checkout(cart))

File: examples/js/01_type_error.js
function totalPrice(items, taxRate) {
  const subtotal = items.reduce((sum, item) => sum + item.price, 0);
  return subtotal + taxRate;
}

function checkout(cart) {
  return totalPrice(cart, "0.08");
}

const cart = [{ price: 10 }, { price: 20 }, { price: 5 }];
const total = checkout(cart);
if (typeof total !== "number" || Number.isNaN(total)) {
  throw new TypeError(`checkout returned non-numeric total: ${total}`);
}
console.log(total);
